packages/block-library/src/post-comments/edit.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useEntityId } from '@wordpress/core-data';
+
+function PostCommentsDisplay( { postId } ) {
+	return useSelect(
+		( select ) => {
+			const comments = select( 'core' ).getEntityRecords( 'root', 'comment', {
+				post: postId,
+			} );
+			return (
+				comments &&
+				comments.map( ( comment ) => <p key={ comment.id }>{ comment.content.raw }</p> )
+			);
+		},
+		[ postId ]
+	);
+}
+
+export default function PostCommentsEdit() {
+	const postId = useEntityId( 'postType', 'post' );
```

Comment:
We may have other post types that support comments, for all these post types we are rendering 'Post Comments Placeholder' while I think we should render the same we render for posts (the comments in that post).

For post types that don't support comments but are not templates or template parts, I think we should not show the block on the inserter e.g: remove it from the allowed blocks or if we should in the inserter we should say in the block that comments are not supported. The block should handle the condition and show this message even if hidden from the inserter because a CPT may have supported comments in the past and then the user may disable the comments.

Reply:
This makes sense, but we'll have to handle it at another level in the code.

It will likely require changes to `core-data` to get post type capabilities and have CPTs act as posts under certain conditions.

Reply:
This is actually a very good remark and one that should lead to some rethinking in how the context EntityProvider and `useEntityProp/useEntityId` works.

Ideally, these hooks should just work no matter the CPT (all CPTs can have all properties) and should not have "post" hard-coded.

My first thinking here is that we should be able to do something like:

```
const entityId = useEntityId( 'postType' );
const entityContent = useEntityProp( 'postType', null, 'content' );
``
`

And these should basically return the first entity in the tree with the "kind" (postType)

We could even imagine a use-case for a post where we'd want any entity (less likely but why not)

```
const entityId = useEntityId();
const entityContent = useEntityProp( null, null, 'content' )
```

Too bad `useEntityProp` is a stable API now though cause "null" for  the first and second args is not great.

Reply:
This is not the first block where it's specific to "post" so we need to solve this ASAP on a follow-up PR.

Reply:
Yes, this is what I suggested.

But, https://github.com/WordPress/gutenberg/issues/19685#issuecomment-576010263 is my preferred alternative that also works on the server.